scripts/compile-sass.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-console */
+
+'use strict';
+
+const path = require('path');
+
+const autoprefixer = require('autoprefixer');
+const fs = require('fs-extra');
+const postcss = require('postcss');
+const sass = require('sass');
+
+const baseDir = path.resolve(__dirname, '..');
+const inDir = path.join(baseDir, 'src', 'scss');
+const outDir = path.join(baseDir, '_built', 'css');
+
+let pnp;
+
+try {
+  // eslint-disable-next-line global-require
+  pnp = require('pnpapi');
+} catch (error) {
+  // not in PnP; not a problem
+}
+
+const resolved = new Map();
+
+const nodeImporter = (url, prev, done) => {
+  if (url.startsWith('~')) {
+    if (pnp) {
+      const components = url.substring(1).split('/');
+      let res;
+      if (resolved.has(components[0])) {
+        res = resolved.get(components[0]);
+      } else {
+        res = pnp.resolveToUnqualified(components[0], `${baseDir}/`);
+        resolved.set(components[0], res);
+      }
+      if (res === null) {
+        done(res);
+      } else {
+        components[0] = res;
+        done({ file: components.join('/') });
+      }
+    } else {
+      done({ file: path.join(baseDir, 'node_modules', url.substring(1)) });
+    }
+  } else {
+    done({ file: url });
+  }
+};
```

Comment:
@mirisuzanne This is essentially the piece that makes this entire script necessary. If the Sass import begins with `~`, then we split the string by `/` (to hopefully get the module name, without nested files), try to resolve the first section as a module name, then update the path to point to the right place (either `node_modules` or `.yarn/cache`).

I should probably do some testing with namespaced packages (e.g. `@oddbird/foobar`)...

Reply:
The watcher seems maybe a bit slower, but otherwise I like `~` better than `../../../node_modules/`. I think the watcher/reload performance is maybe a separate question for you. :) Sass changes trigger a slow double-reload, which isn't new in this branch.